### PR TITLE
Add accept terms alert.

### PR DESCRIPTION
**Pull request by roberthein**
img_4063
img_4064

### CODE REVIEW

File: Toshi/Controllers/Sign In/SplashViewController.swift [hunk 1]
```diff
@@ -133,11 +134,44 @@ final class SplashViewController: UIViewController {
...
     
     open override var preferredStatusBarStyle: UIStatusBarStyle {
         return .lightContent
     }
+    
+    fileprivate func showAcceptTermsAlert() {
+        
+        let alert = UIAlertController(title: Localized("accept_terms_title"), message: Localized("accept_terms_text"), preferredStyle: .alert)
+        
+        let read = UIAlertAction(title: Localized("accept_terms_action_read"), style: .default) { [weak self] _ in
```
Comment (yuliaveres) [inline Toshi/Controllers/Sign In/SplashViewController.swift]: the `self` is not captured by alert action so you might omit weak self 👍  not required though

File: Toshi/Controllers/Sign In/SplashViewController.swift [hunk 2]
```diff
@@ -133,11 +134,44 @@ final class SplashViewController: UIViewController {
...
+            let controller = SFSafariViewController(url: url, entersReaderIfAvailable: true)
+            controller.delegate = self
+            self?.present(controller, animated: true, completion: nil)
+        }
+        
+        let cancel = UIAlertAction(title: Localized("accept_terms_action_cancel"), style: .default) { _ in
+            alert.dismiss(animated: true, completion: nil)
+        }
+        
+        let agree = UIAlertAction(title: Localized("accept_terms_action_agree"), style: .default) { [weak self] _ in
```
Comment (yuliaveres) [inline Toshi/Controllers/Sign In/SplashViewController.swift]: `agree` i guess might be highlighted, do you mind making it Cancel style ?

File: Toshi/Controllers/Sign In/SplashViewController.swift [hunk 3]
```diff
@@ -133,11 +134,44 @@ final class SplashViewController: UIViewController {
...
     }
+    
+    fileprivate func showAcceptTermsAlert() {
+        
+        let alert = UIAlertController(title: Localized("accept_terms_title"), message: Localized("accept_terms_text"), preferredStyle: .alert)
+        
+        let read = UIAlertAction(title: Localized("accept_terms_action_read"), style: .default) { [weak self] _ in
+            guard let url = URL(string: "http://www.toshi.org/terms-of-service/") else { return }
+            let controller = SFSafariViewController(url: url, entersReaderIfAvailable: true)
+            controller.delegate = self
```
Comment (yuliaveres) [inline Toshi/Controllers/Sign In/SplashViewController.swift]: have u tried making `preferredControlTintColor` to our theme tint one  ?

File: Toshi/Controllers/Sign In/SplashViewController.swift (hunk 1 above)
Comment (roberthein) [reply Toshi/Controllers/Sign In/SplashViewController.swift]: But I'm using:
```
self?.present(controller, animated: true, completion: nil)
```
So it is captured, right?

File: Toshi/Controllers/Sign In/SplashViewController.swift (hunk 2 above)
Comment (roberthein) [reply Toshi/Controllers/Sign In/SplashViewController.swift]: Perfect!

File: Toshi/Controllers/Sign In/SplashViewController.swift (hunk 3 above)
Comment (roberthein) [reply Toshi/Controllers/Sign In/SplashViewController.swift]: Nice! Much better. 🥇

File: Toshi/Controllers/Sign In/SplashViewController.swift (hunk 1 above)
Comment (yuliaveres) [reply Toshi/Controllers/Sign In/SplashViewController.swift]: nope, it should not be captured here :)

Comment (roberthein) [reply Toshi/Controllers/Sign In/SplashViewController.swift]: So I am calling `self` in a `completion block` but I should not weakify `self`?

**Pull request merged**